test/Concurrency/custom_executor_enqueue_availability.swift

```diff
@@ -0,0 +1,67 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-move-only
+// REQUIRES: concurrency
+// REQUIRES: OS=macosx
+
```

Comment:
This is probably missing a bunch of disables, like:

```suggestion
// UNSUPPORTED: back_deployment_runtime
// REQUIRES: concurrency_runtime
// UNSUPPORTED: freestanding

```

Reply:
The requirements here are based on the ones in custom_executor_enqueue_impls.swift, the primary difference being the removal of `-disable-availability-checking` from the swiftc invocation. Do we need more disables here than in the other test? I see that I missed copying over the `UNSUPPORTED: freestanding` so I'll add that in — but not quite sure why this wouldn't support a back-deployed runtime.

Reply:
we don't backdeploy the `ExecutorJob` type, so since the test is touching that I don't think it'll work there

Reply:
All of the usages of `ExecutorJob` are annotated with `@available(SwiftStdlib 5.9, *)` — that seems sufficient to me but correct me if I'm missing something. The CI checks seem to have passed, is there a specific configuration they aren't covering that I can test manually?